subgraphs/rocket-pool/src/checkpoints/nodeUtilities.ts

```diff
@@ -0,0 +1,382 @@
+import { ethereum, BigInt, BigDecimal } from "@graphprotocol/graph-ts";
+import {
+  NetworkNodeBalanceCheckpoint,
+  Node,
+  RocketPoolProtocol,
+} from "../../generated/schema";
+import { generalUtilities } from "./generalUtilities";
+import { NetworkNodeBalanceMinipoolMetadata } from "../models/networkNodeBalanceMinipoolMetadata";
+import { ONE_ETHER_IN_WEI } from "../constants/generalConstants";
+import { NetworkNodeBalanceRPLMetadata } from "../models/networkNodeBalanceRPLMetadata";
+import { updateProtocolAndPoolRewardsTvl } from "../updaters/financialMetrics";
+
+class NodeUtilities {
+  /**
+   * Checks if there is already an indexed network node balance checkpoint for the given event.
+   */
+  public hasNetworkNodeBalanceCheckpointHasBeenIndexed(
+    protocol: RocketPoolProtocol,
+    event: ethereum.Event
+  ): boolean {
+    // If this specific event has been handled, then return true.
+    if (
+      NetworkNodeBalanceCheckpoint.load(
+        generalUtilities.extractIdForEntity(event)
+      ) !== null
+    )
+      return true;
+
+    // No indexed protocol means there is no latest network node balance checkpoint.
+    if (protocol === null) return false;
+
+    /*
+    Retrieve the latest network balance checkpoint.
+    If there is none at the moment, return false because this hasnt been handled yet.
+  */
+    const latestNetworkNodeBalanceCheckpointId =
+      protocol.lastNetworkNodeBalanceCheckPoint;
+    if (latestNetworkNodeBalanceCheckpointId === null) return false;
+    const latestNetworkNodeBalanceCheckpoint =
+      NetworkNodeBalanceCheckpoint.load(latestNetworkNodeBalanceCheckpointId);
+    if (
+      latestNetworkNodeBalanceCheckpoint === null ||
+      latestNetworkNodeBalanceCheckpoint.blockTime == BigInt.fromI32(0)
+    )
+      return false;
+
+    // Get the date of the network node balance event candidate and the latest network node balance checkpoint.
+    const dateOfNewNetworkNodeBalanceCheckpoint = new Date(
+      event.block.timestamp.toI32() * 1000
+    );
+    const dateOfLatestNetworkNodeBalanceCheckpoint = new Date(
+      latestNetworkNodeBalanceCheckpoint.blockTime.toI32() * 1000
+    );
+
+    // If the latest network node balance checkpoint and the candidate match in terms of day/month/year, then return false.
+    return (
```

Comment:
Could you take the `timestamp / SecondsPerDay` and compare to get the same result?

Reply:
Maybe, but not sure why we'd make this change to be less precise when the code being replaced is O(1) / alr loaded into memory by function

Reply:
fair enough, resolved 👌